### PR TITLE
NAD-1089 docs(select): add design ref tab

### PR DESCRIPTION
**Pull request by kevinch**
#### What is the purpose of this pull request?
Add design reference tab content.

#### What problem is this solving?
Lack of content.

#### How should this be manually tested?
See links below.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui-docs/docs/guidelines/components/select.mdx [hunk 1]
```diff
@@ -105,3 +122,19 @@ Besides the props listed bellow, you can use all the props accepted by the `sele
...
 | criticalText | `string`            | TextArea critical message | 🚫       | -         |
 | csx          | `StyleProp`         | Defines component styles  | 🚫       | `{}`      |
+
+</Panel>
+<Panel id="designReference">
+
+## Do’s
+
+1. Mind the order of the options, like putting the more probable one to be selected at the top. In doubt, sort them alphanumerically (from A to Z and from 0 to 9).
+2. If there are more than four options listed, consider using a radio button.
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/guidelines/components/select.mdx:132]: less than

File: packages/admin-ui-docs/docs/guidelines/components/select.mdx [hunk 2]
```diff
@@ -105,3 +122,19 @@ Besides the props listed bellow, you can use all the props accepted by the `sele
...
+3. Always leave an option from the list selected by default.
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/guidelines/components/select.mdx:133]: This third point can be removed

File: packages/admin-ui-docs/docs/guidelines/components/select.mdx [hunk 3]
```diff
@@ -105,3 +122,19 @@ Besides the props listed bellow, you can use all the props accepted by the `sele
...
+4. Don't offer overlapping options, they should be clearly mutually exclusive.
+5. Indicate the chosen, or default option with a “check” icon.
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/guidelines/components/select.mdx:135]: This fifth point can be removed

**Review by davicostalf — CHANGES_REQUESTED** [review]

File: packages/admin-ui-docs/docs/guidelines/components/select.mdx (hunk 1 above)
Comment (kevinch) [reply packages/admin-ui-docs/docs/guidelines/components/select.mdx:132]: How so?

Comment (davicostalf) [reply packages/admin-ui-docs/docs/guidelines/components/select.mdx:132]: instead of "more than" it should be "less than"

**Review by davicostalf — CHANGES_REQUESTED** [review]

File: packages/admin-ui-docs/docs/guidelines/components/select.mdx (hunk 1 above)
Comment (marcelovicentegc) [reply packages/admin-ui-docs/docs/guidelines/components/select.mdx:132]: Done!

**Pull request merged**